viz-lib/src/visualizations/chart/plotly/updateYRanges.js

```diff
@@ -4,6 +4,42 @@ function calculateAxisRange(range, min, max) {
   return [isNumber(min) ? min : range[0], isNumber(max) ? max : range[1]];
 }
 
+function calculateAbsoluteDiff(value, totalRange, percentageDiff) {
+  return (percentageDiff * totalRange) / (1 - Math.abs(value) / totalRange - percentageDiff);
+}
+
+function alignYAxesAtZero(axisA, axisB) {
+  // Make sure the origin is included in both axes
+  axisA.range[1] = Math.max(0, axisA.range[1]);
+  axisB.range[1] = Math.max(0, axisB.range[1]);
+  axisA.range[0] = Math.min(0, axisA.range[0]);
+  axisB.range[0] = Math.min(0, axisB.range[0]);
```

Comment:
That's actually very interesting piece. We can align axes at 0 when both (or maybe at leat one) of them cross 0 point. But if both have ranges that are completely above/below zero - should we really do anything? That's a good question

Reply:
For me it works both with making sure the origin is included or with checking if it is included before. I.e.: You can add a guarding `if (!inRange(a) && !inRange(b)) return;` before everything at will.

Note: the origin must be included in both axes before running the code below, otherwise it can have negative values in the calculation, causing some weird results (basically it inverts the "by always augmenting").